osu.Game.Rulesets.Osu/Edit/PolygonGenerationPopover.cs

```diff
@@ -136,39 +138,58 @@ private void tryCreatePolygon()
             double length = distanceSnapInput.Current.Value * velocity * timeSpacing;
             float polygonRadius = (float)(length / (2 * Math.Sin(double.Pi / pointInput.Current.Value)));
 
-            editorBeatmap.RemoveRange(insertedCircles);
-            insertedCircles.Clear();
+            int totalPoints = pointInput.Current.Value * repeatCountInput.Current.Value;
+
+            if (insertedCircles.Count > totalPoints)
+            {
+                editorBeatmap.RemoveRange(insertedCircles.GetRange(totalPoints + 1, insertedCircles.Count - totalPoints - 1));
+                insertedCircles.RemoveRange(totalPoints + 1, insertedCircles.Count - totalPoints - 1);
```

Comment:
This is off by one.

To illustrate with an example: Say you have 10 points and want to leave 5. Thus, you want to keep 5, starting from index 0 through 4, and delete 5, starting from index 5 through 9. So the `+ 1` everywhere is gratuitous / wrong.

```suggestion
                editorBeatmap.RemoveRange(insertedCircles.GetRange(totalPoints, insertedCircles.Count - totalPoints));
                insertedCircles.RemoveRange(totalPoints, insertedCircles.Count - totalPoints);
```

You can actually see this fail from the default settings if you increase the number of vertices by 1 and then decrease it again - 4 objects will get created, not 3.

(This is partially why I was asking as to how much performance this was actually gaining, because this sort of logic is very easy to get wrong and it's just simpler to not have to think about it if it's not necessary.)

Reply:
Oh, yeah that's a silly mistake on my part.